src/mono/wasm/debugger/DebuggerTestSuite/EvaluateOnCallFrameTests.cs

```diff
@@ -1178,47 +1178,67 @@ async Task EvaluateOnCallFrameFail(string call_frame_id, params (string expressi
+                    await CheckValue(actualValObj["value"], expectedListRootHiddenElements[i]["value"], "listRootHidden");
                 }
-                foreach (var item in refArrayProp)
+
+                for (int i = 0; i < expectedArrayElements.Length; i ++)
                 {
-                    item["name"] = string.Concat("arrayRootHidden[", item["name"], "]");
-                    CheckContainsJObject(testRootHiddenProps, item, item["name"].Value<string>());
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, expectedArrayElements[i]["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], expectedArrayElements[i]["value"], "x");
                 }
 
-                // valuetype/class members unique names are created by concatenation with a dot
-                foreach (var item in refStructProp)
+                foreach (var structItem in expectedStructRootHiddenElements)
                 {
-                    item["name"] = string.Concat("sampleStructRootHidden.", item["name"]);
-                    CheckContainsJObject(testRootHiddenProps, item, item["name"].Value<string>());
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, structItem["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], structItem["value"], "sampleStructRootHidden");
                 }
-                foreach (var item in refClassProp)
+
+                foreach (var classItem in expectedClassRootHiddenElements)
                 {
-                    item["name"] = string.Concat("sampleClassRootHidden.", item["name"]);
-                    CheckContainsJObject(testRootHiddenProps, item, item["name"].Value<string>());
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, classItem["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], classItem["value"], "sampleClassRootHidden");
                 }
+
+                JObject[] expectedTestRootHiddenProps = new[]
+                {
+                    expectedArrayElements[0],
+                    expectedArrayElements[1],
+                    expectedListRootHiddenElements[0],
+                    expectedListRootHiddenElements[1],
+                    expectedClassRootHiddenElements[0],
+                    expectedClassRootHiddenElements[1],
+                    expectedStructRootHiddenElements[0],
+                    expectedStructRootHiddenElements[1]
+                };
+                // CheckProps for JArrays care about the order and here the order differs between InlineData
+                var testRootHiddenPropsSorted = new JArray(testRootHiddenProps.OrderBy(v => (string)v["name"]));
+                await CheckProps(testRootHiddenPropsSorted, expectedTestRootHiddenProps, "listRootHidden");
```

Comment:
If we are checking all the members here, then why are the earlier loops needed?

Reply:
Not needed, `CheckValue` calls `CheckProps`, so we can shorten this way by leaving only the quoted lines.